### ISSUE TITLE
[bug] useAccount onConnect onDisconnect

### ISSUE THREAD

**Issue by soccerarenadev** [wagmi-dev/wagmi#777]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.5.12 and 0.6.0

### Current Behavior

Using this code:

```
    const accountInfo = useAccount({
      onConnect: ({ address, connector, isReconnected }) => {
        console.log('Connected', { address, connector, isReconnected })
      },
      onDisconnect: () => {
        console.log('Disconnected')
      },
     }
    )

```
I'm getting the "connected" and "disconnected" log randomly. I haven't found a pattern. I can reload the web and works, and reload again and stops working.

### Expected Behavior

Always trigger onConnect and onDisconnect

**Comment by tmm** [comment]
Thanks for reporting. This issue is temporarily closed due to lack of information. Please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) and reopen this issue.

**Comment by soccerarenadev** [comment]
NextJS Page:

```
import { getCsrfToken, signIn, signOut, useSession } from "next-auth/react"
import { ConnectButton, RainbowKitProvider } from '@rainbow-me/rainbowkit';
import { useAccount, useBlockNumber, useConnect, useDisconnect, useNetwork, useSignMessage } from 'wagmi'



const MinimalError = () => {
    const { data: authjsSession, status: authjsStatus } = useSession();
    const signMessage = useSignMessage( )
    const { disconnect } = useDisconnect()
    const { chain: activeChain } = useNetwork()
    const accountInfo = useAccount({
      onConnect: ({ address, connector, isReconnected }) => {
        console.error('Connected', { address, connector, isReconnected })
      },
      onDisconnect: () => {
        console.error('Disconnected')
      },
     }
    )

  return ( <ConnectButton /> );
};

export default MinimalError;
```


_app.tsx:
```
import { appWithTranslation } from 'next-i18next';
import { AppProps } from 'next/app';

import { RainbowKitProvider, getDefaultWallets } from '@rainbow-me/rainbowkit';

import { chain, configureChains, createClient, WagmiConfig } from 'wagmi';
import { alchemyProvider } from 'wagmi/providers/alchemy';
import { publicProvider } from 'wagmi/providers/public';
import '@rainbow-me/rainbowkit/styles.css';
import { SessionProvider } from 'next-auth/react';

const { chains, provider, webSocketProvider } = configureChains(
  [
    chain.mainnet,
    chain.polygon,
  ],
  [
    alchemyProvider({
      // This is Alchemy's default API key.
      // You can get your own at https://dashboard.alchemyapi.io
      alchemyId: '_gg7wSSi0KMBsdKnGVfHDueq6xMB9EkC',
    }),
    publicProvider(),
  ]
);


const { connectors } = getDefaultWallets({
  appName: 'appname',
  chains,
});

const wagmiClient = createClient({
  autoConnect: false,
  connectors,
  provider,
  webSocketProvider,
});


const MyApp = ({ Component, pageProps }: AppProps) =>  {
  return (
    <WagmiConfig client={wagmiClient}>
      <SessionProvider session={pageProps.session} refetchInterval={0}>
        <RainbowKitProvider chains={chains}>
          <Component {...pageProps} />
        </RainbowKitProvider>
      </SessionProvider>
    </WagmiConfig>
  )
};

export default appWithTranslation(MyApp);
```

**Comment by wong2** [comment]
same here

**Comment by reecejohnson** [comment]
I'm also seeing the same issue. 

The callbacks `onConnect` and `onDisconnect` are  intermittently called.

**Comment by KristjanBajuk** [comment]
I have the same issue

**Comment by Quix44** [comment]
bump! also have this..

**Comment by tmm** [comment]
@Quix44 this was fixed in [`wagmi@0.7.4`](https://github.com/wagmi-dev/wagmi/releases/tag/wagmi%400.7.4).

**Comment by NBMSacha** [comment]
Nope @tmm, still getting this issue with wagmi@0.8.6 :/
